File: src/main/java/club/xiaojiawei/bean/EntityMapper.java
package club.xiaojiawei.bean;

import club.xiaojiawei.bean.entity.Card;
import club.xiaojiawei.bean.entity.Entity;
import org.mapstruct.Mapper;
import org.mapstruct.MappingTarget;
import org.mapstruct.factory.Mappers;

/**
 * @author 肖嘉威 xjw580@qq.com
 * @date 2024/1/6 18:38
 */
@Mapper
public interface EntityMapper {

    EntityMapper INSTANCE = Mappers.getMapper(EntityMapper.class);

    void update(Entity sourceEntity, @MappingTarget Entity targetEntity);

}

File: hs-script-card-sdk/src/main/kotlin/club/xiaojiawei/mapper/EntityMapper.kt
package club.xiaojiawei.mapper

import club.xiaojiawei.bean.Entity
import org.mapstruct.Mapper
import org.mapstruct.MappingTarget
import org.mapstruct.factory.Mappers

/**
 * @author 肖嘉威
 * @date 2024/9/8 19:17
 */
@Mapper
interface EntityMapper {

    companion object {
        val INSTANCE: EntityMapper = Mappers.getMapper(EntityMapper::class.java)
    }

    fun update(sourceEntity: Entity?, @MappingTarget targetEntity: Entity?)

}